mesonbuild/interpreter/dependencyfallbacks.py

```diff
@@ -0,0 +1,346 @@
+                         'found:', mlog.red('NO'), *info)
+                return DependencyHolder(cached_dep, self.subproject)
+        else:
+            info = [mlog.blue('(cached)')]
+            cached_dep = self.coredata.deps[for_machine].get(identifier)
+
+        if cached_dep:
+            found_vers = cached_dep.get_version()
+            if not self._check_version(wanted_vers, found_vers):
+                mlog.log('Dependency', mlog.bold(name),
+                         'found:', mlog.red('NO'),
+                         'found', mlog.normal_cyan(found_vers), 'but need:',
+                         mlog.bold(', '.join([f"'{e}'" for e in wanted_vers])),
+                         *info)
+                return self._notfound_dependency()
+            if found_vers:
+                info = [mlog.normal_cyan(found_vers), *info]
+            mlog.log('Dependency', mlog.bold(self.display_name),
+                     'found:', mlog.green('YES'), *info)
+            return DependencyHolder(cached_dep, self.subproject)
+        return None
+
+    def _get_subproject_variable(self, subproject: SubprojectHolder, varname: str) -> T.Optional[DependencyHolder]:
+        var_dep = subproject.held_object.variables.get(varname)
+        if not isinstance(var_dep, DependencyHolder):
+            mlog.warning(f'Variable {varname!r} in the subproject {subproject.subdir!r} is',
+                         'not found' if var_dep is None else 'not a dependency object')
+            return None
+        return var_dep
+
+    def _verify_fallback_consistency(self, cached_dep: DependencyHolder):
+        subp_name = self.subproject_name
+        varname = self.subproject_varname
+        subproject = self._get_subproject(subp_name)
+        if subproject and varname:
+            var_dep = self._get_subproject_variable(subproject, varname)
+            if var_dep and cached_dep.found() and var_dep.held_object != cached_dep.held_object:
+                mlog.warning(f'Inconsistency: Subproject has overridden the dependency with another variable than {varname!r}')
+
+    def _handle_featurenew_dependencies(self, name: str) -> None:
+        'Do a feature check on dependencies used by this subproject'
+        if name == 'mpi':
+            FeatureNew.single_use('MPI Dependency', '0.42.0', self.subproject)
+        elif name == 'pcap':
+            FeatureNew.single_use('Pcap Dependency', '0.42.0', self.subproject)
+        elif name == 'vulkan':
+            FeatureNew.single_use('Vulkan Dependency', '0.42.0', self.subproject)
+        elif name == 'libwmf':
+            FeatureNew.single_use('LibWMF Dependency', '0.44.0', self.subproject)
+        elif name == 'openmp':
+            FeatureNew.single_use('OpenMP Dependency', '0.46.0', self.subproject)
+
+    def _notfound_dependency(self) -> DependencyHolder:
+        return DependencyHolder(NotFoundDependency(self.environment), self.subproject)
```

Comment:
A function for `DependencyHolder(NotFoundDependency(self.environment), self.subproject)` is a bit overkill IMHO.

Especially, since the `DependencyHolder` part will be gone after #8885 is merged.

Reply:
It was already there in previous code, I think it's nicer tbh. Will be easier for your refactoring, just one place to edit instead of everywhere it's called.

Reply:
Sure, it is just a very minor style choice anyway.